tensorflow/core/kernels/cwise_op_gpu_select.cu.cc

```diff
@@ -33,6 +33,20 @@ struct SelectFunctor<GPUDevice, T> {
 };
 
 template <typename T>
+struct SelectScalarFunctor<GPUDevice, T> {
+  void operator()(const GPUDevice& d, typename TTypes<T>::Flat out,
+                  typename TTypes<bool>::ConstScalar cond,
+                  typename TTypes<T>::ConstFlat then_flat,
+                  typename TTypes<T>::ConstFlat else_flat) {
+    if (cond()){
```

Comment:
In this case you want to broadcast the scalar value to a vector and pass
the ref to the standard select functor.  I'd like to see that there's a
performance benefit to not copying a single scalar from GPU to CPU though,
to justify all this.

On Thu, Sep 8, 2016 at 9:07 AM, Vijay Vasudevan <notifications@github.com>
wrote:

> In tensorflow/core/kernels/cwise_op_gpu_select.cu.cc
> <https://github.com/tensorflow/tensorflow/pull/3995#discussion_r78037075>:
>
> > @@ -33,6 +33,20 @@ struct SelectFunctor<GPUDevice, T> {
> >  };
> >
> >  template <typename T>
> > +struct SelectScalarFunctor<GPUDevice, T> {
> > +  void operator()(const GPUDevice& d, typename TTypes<T>::Flat out,
> > +                  typename TTypes<bool>::ConstScalar cond,
> > +                  typename TTypes<T>::ConstFlat then_flat,
> > +                  typename TTypes<T>::ConstFlat else_flat) {
> > +    if (cond()){
>
> I think the whole point of this op was to have the 'cond' be on GPU
> though. Otherwise you might as well just use tf.cond()
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/3995/files/60290e35ee2b53e7ea07b62bf344d0b93ee5b505#r78037075>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABtimwwJPuLKTZiS-__jUUwyvq_X6aU-ks5qoDLWgaJpZM4JrZh->
> .
>